torch/optim/adamax.py

```diff
@@ -83,6 +94,9 @@ def step(self, closure=None):
                 bias_correction = 1 - beta1 ** state['step']
                 clr = group['lr'] / bias_correction
 
+                if group['weight_decay'] != 0 and not group['l2_reg']:
+                    p.data.add_(-group['weight_decay'], p.data)
```

Comment:
```suggestion
                    p.data.add_(-group['weight_decay'] * group['lr'] , p.data)
```

In L2, the weight decay is stored inside the grad variable, which is later multiplied by the learning rate. This new formulation is not multiplying it by the learning rate (see Algorithms 1 and 2 in the paper).